colorcoder/Main.java

```diff
@@ -1,120 +1,29 @@
-            this.index = index;
-        }
-        int getIndex() {
-            return index;
-        }
-        public static MinorColor fromIndex(int index) {
-            for(MinorColor color: MinorColor.values()) {
-                if(color.getIndex() == index) {
-                    return color;
-                }
-            }
-            return null;
-        }
-    };
-
-    final static String MajorColorNames[] = {
-        "White", "Red", "Black", "Yellow", "Violet"
-    };
-    final static int numberOfMajorColors = MajorColorNames.length;
-    final static String MinorColorNames[] = {
-        "Blue", "Orange", "Green", "Brown", "Slate"
-    };
-    final static int numberOfMinorColors = MinorColorNames.length;
+import java.util.Map;
 
-    static class ColorPair {
-        private MajorColor majorColor;
-        private MinorColor minorColor;
-        
-        public ColorPair(MajorColor major, MinorColor minor)
-        {
-            majorColor = major;
-            minorColor = minor;
-        }
-        public MajorColor getMajor() {
-            return majorColor;
-        }
-        public MinorColor getMinor() {
-            return minorColor;
-        }
-        String ToString() {
-            String colorPairStr = MajorColorNames[majorColor.getIndex()];
-            colorPairStr += " ";
-            colorPairStr += MinorColorNames[minorColor.getIndex()];
-            return colorPairStr;
-        }
-    };
+public class Main {
 
-    static ColorPair GetColorFromPairNumber(int pairNumber) {
-        int zeroBasedPairNumber = pairNumber - 1;
-        MajorColor majorColor = 
-            MajorColor.fromIndex(zeroBasedPairNumber / numberOfMinorColors);
-        MinorColor minorColor =
-            MinorColor.fromIndex(zeroBasedPairNumber % numberOfMinorColors);
-        return new ColorPair(majorColor, minorColor);
-    }
-    static int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
-        return major.getIndex() * numberOfMinorColors + minor.getIndex() + 1;
-    }
+    public static void main(String[] args) {
 
-    static void testNumberToPair(int pairNumber,
-        MajorColor expectedMajor,
-        MinorColor expectedMinor)
-    {
-        ColorPair colorPair = GetColorFromPairNumber(pairNumber);
-        System.out.println("Got pair " + colorPair.ToString());
-        assert(colorPair.getMajor() == expectedMajor);
-        assert(colorPair.getMinor() == expectedMinor);
-    }
+        Main main = new Main();
+        ColorPairHelper.testNumberToPair(4, MajorColor.WHITE, MinorColor.BROWN);
+        ColorPairHelper.testNumberToPair(5, MajorColor.WHITE, MinorColor.SLATE);
+        ColorPairHelper.testPairToNumber(MajorColor.BLACK, MinorColor.ORANGE, 12);
+        ColorPairHelper.testPairToNumber(MajorColor.VIOLET, MinorColor.SLATE, 25);
+        Map<Integer,ColorPair> colorCodingPairMap = ColorPairUtil.getColorCodingPairMap();
+        main.printColorCodingReferenceManual(colorCodingPairMap);
```

Comment:
Printing of colorCoding ref manual is also a part of the production code. But how is the functionality verified? Test case needs to be added to verify the output. Even if we cannot verify what is printed, test case can be added to to check if the list is constructed clearly. Please add such test case.